xen/drivers/passthrough/arm/ipmmu-vmsa.c

```diff
@@ -2588,6 +2616,60 @@ static void ipmmu_vmsa_dump_p2m_table(struct domain *d)
 	/* TODO: This platform callback should be implemented. */
 }
 
+#ifdef CONFIG_HAS_COPROC
+static inline int ipmmu_vmsa_assign_coproc(struct domain *d, struct device *dev)
+{
+	return ipmmu_vmsa_assign_dev(d, 0, dev, 0);
+}
+
+static inline int ipmmu_vmsa_deassign_coproc(struct domain *d, struct device *dev)
+{
+	return ipmmu_vmsa_deassign_dev(d, dev);
+}
+
+static int ipmmu_vmsa_disable_coproc(struct domain *d, struct device *dev)
+{
+	struct ipmmu_vmsa_archdata *archdata;
+	unsigned int i;
+
+	if (!dev_iommu(dev) || !dev_iommu(dev)->is_coproc)
```

Comment:
I don't see any reason.
1.This is not a function this is a macros resulting in field in a structure.
2.We don't work with it down the function, this is a sanity check only.